Merge 2.34 code freeze

mokagio opened the pull request:
Took a few additional commits on top of https://github.com/Automattic/simplenote-android/pull/1688 to get the release build to work, but [got there in the end](https://buildkite.com/automattic/simplenote-android/builds/365#_):

![image](https://github.com/user-attachments/assets/971088c6-b45d-48f2-8058-3a8e494ad29a)

<img width="685" alt="image" src="https://github.com/user-attachments/assets/48a28624-ee9e-4254-bb29-6eb6a83b731b">

notandyvee committed:
[Bug] If a deeplink is malformed or missing data we show auth screen.

notandyvee committed:
Show loading dialog when loading magic link.

wpmobilebot committed:
Update strings file for translation automation

wpmobilebot committed:
Bump version name and build code

wpmobilebot committed:
Update draft release notes for 2.34.

wpmobilebot committed:
Release Notes: add new section for next version (2.35)

mokagio committed:
Update release notes source with placeholder copy for 2.34

mokagio committed:
Merge remote-tracking branch 'origin/trunk' into andy/firebird-issue-81

mokagio committed:
Update release notes source with placeholder copy for 2.34 (#1691)

mokagio committed:
Use `RELEASE_VERSION` instead of `BUILDKITE_RELEASE_VERSION`

The `BUILDKITE_` prefix is a workaround required only on iOS.
We could adopt it for consistency here, too, but it's best to avoid it
for two reasons:

1. Being a workaround, one would hope to eventually remove it
2. The less coupled we are with the particular CI implementation details
   (i.e. "Buildkite" in the name) the better

mokagio committed:
Update `.configure` using latest toolkit version

Notice how there now is an explicit `encrypt: true` for each file.

wpmobilebot committed:
Update `PlayStoreStrings.pot` for 2.34

mokagio committed:
Fix `BUILDKITE_ORGANIZATION` defined as `BUILDKITE_ORG`

This was the reason for the failure in
https://buildkite.com/automattic/simplenote-android/builds/354#0191db6f-e529-425f-b50e-c62154395e01

mokagio committed:
Add missing `version` and `build_code` to `build_for_distribution`

mokagio committed:
Account or different globbing behavior in Bash vs Zsh

Zsh is the shell I use day to day and on which I tried the commands I
put in the script.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR has the <code>Releases...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR has the <code>Releases</code> label: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Specify `-maxdepth` before `-name`

See warning from
https://buildkite.com/automattic/simplenote-android/builds/361#0191deaf-ee5a-4c77-8e48-2781416d93cd

[2024-09-11T01:32:31Z] find: warning: you have specified the global
option -maxdepth after the argument -name, but global options are not
positional, i.e., -maxdepth affects tests specified before it as well as
those specified after it.  Please specify global options before other
arguments.

mokagio committed:
Make all release Buildkite command scripts executable

See how https://buildkite.com/automattic/simplenote-android/builds/361
failed

It's weird, though, that `release-build.sh` was not executable yet it
worked. Maybe it has something to do with the fact that it's called
without arguments? Might it be a Buildkite implementation detail?

mokagio committed:
Fix incorrect APK path in upload script

mokagio committed:
Use Fastlane to decrypt secrets in prototype build step

This is a followup to 6fce8c8c39968709b7923c0ad96b091b2bd7dc9a

Unfortunately, we need to use the same configure implementation across
the whole CI automation due to existing compatibility bugs between the
Ruby and Rust versions

mokagio committed:
Use full release name when creating GitHub release

AliSoftware commented on .buildkite/commands/checkout-release-branch.sh:
```diff
@@ -1,14 +1,12 @@
-# It must use the `BUILDKITE_` prefix to be passed to the agent due to how `hostmgr` works, in case this runs on a Mac agents.
-if [[ -z "${BUILDKITE_RELEASE_VERSION}" ]]; then
-    echo "BUILDKITE_RELEASE_VERSION is not set."
+if [[ -z "${RELEASE_VERSION}" ]]; then
```
I double-checked and this `checkout-release-branch.sh` script seems to only be called by `complete-code-freeze.yml`, from a step that runs on `tumblr-metal`.

So indeed there's no risk of the env var being filtered out by `hostmgr` when the script to run is passed from Mac host to VM here, because this will run on `tumblr-metal` and not Mac agents, so `hostmgr` is not even involved in the first place. 👍

notandyvee committed:
[Bug] If a deeplink is malformed or missing data we show auth screen. (#1673)
